src/profile-logic/marker-data.js

```diff
@@ -0,0 +1,318 @@
+    const { data, name } = marker;
+    const lowerCaseName = name.toLowerCase();
+    if (lowerCaseName.includes(lowerCaseSearchString)) {
+      newMarkers.push(marker);
+      continue;
+    }
+    if (data && typeof data === 'object') {
+      if (
+        typeof data.eventType === 'string' &&
+        data.eventType.toLowerCase().includes(lowerCaseSearchString)
+      ) {
+        // Match DOMevents data.eventType
+        newMarkers.push(marker);
+        continue;
+      }
+      if (
+        typeof data.name === 'string' &&
+        data.name.toLowerCase().includes(lowerCaseSearchString)
+      ) {
+        // Match UserTiming's name.
+        newMarkers.push(marker);
+        continue;
+      }
+      if (
+        typeof data.category === 'string' &&
+        data.category.toLowerCase().includes(lowerCaseSearchString)
+      ) {
+        // Match UserTiming's name.
+        newMarkers.push(marker);
+        continue;
+      }
+    }
+  }
+  return newMarkers;
+}
+
+/**
+ * This function takes a marker that packs in a marker payload into the string of the
+ * name. This extracts that and turns it into a payload.
+ */
+export function extractMarkerDataFromName(
+  markers: MarkersTable,
+  stringTable: UniqueStringArray
+): MarkersTable {
+  const newMarkers: MarkersTable = {
+    data: markers.data.slice(),
+    name: markers.name.slice(),
+    time: markers.time.slice(),
+    length: markers.length,
+  };
+
+  // Match: "Bailout_MonitorTypes after add on line 1013 of self-hosted:1008"
+  // Match: "Bailout_TypeBarrierO at jumptarget on line 1490 of resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:1484"
+  const bailoutRegex =
+    // Capture groups:
+    //       type   afterAt    where        bailoutLine  script functionLine
+    //        ↓     ↓          ↓                  ↓        ↓    ↓
+    /^Bailout_(\w+) (after|at) ([\w _-]+) on line (\d+) of (.*):(\d+)$/;
+
+  // Match: "Invalidate resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:3662"
+  // Match: "Invalidate self-hosted:4032"
+  const invalidateRegex =
+    // Capture groups:
+    //         url    line
+    //           ↓    ↓
+    /^Invalidate (.*):(\d+)$/;
+
+  for (let markerIndex = 0; markerIndex < markers.length; markerIndex++) {
+    const nameIndex = markers.name[markerIndex];
+    const time = markers.time[markerIndex];
+    const name = stringTable.getString(nameIndex);
+    let matchFound = false;
+    if (name.startsWith('Bailout_')) {
+      matchFound = true;
+      const match = name.match(bailoutRegex);
+      if (!match) {
+        console.error(`Could not match regex for bailout: "${name}"`);
+      } else {
+        const [
+          ,
+          type,
+          afterAt,
+          where,
+          bailoutLine,
+          script,
+          functionLine,
+        ] = match;
+        newMarkers.name[markerIndex] = nameIndex;
```

Comment:
I think we want to keep what was before: `Bailout` and `Invalidation` instead of the full name. 
I realize you did this so that we still see the full name in the marker table, but this makes the marker chart messy, see [this example](https://deploy-preview-1227--perf-html.netlify.com/public/280e10708c0660695ef5af145c53179d29d9796c/marker-chart/?globalTrackOrder=0-1-2-3-4&localTrackOrderByPid=31626-0~&thread=4&v=3).

A possible solution is to use `title` for this -- but we're too early in the process here because we don't have tracing markers yet. So maybe this step should come after we convert the markers to tracing markers, so that we can have both `name` and `title`, and use `title` in the marker table if non-null.

Reply:
Ah, I didn't realize this would affect the marker chart. Thanks for pointing this out, I will revert it.